packages/angular/src/modules/waas/http-transport-adapter.service.spec.ts

```diff
@@ -0,0 +1,150 @@
+import 'core-js/es7/reflect';
+import 'zone.js/dist/zone';
+import { TestBed } from '@angular/core/testing';
+import { BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting } from '@angular/platform-browser-dynamic/testing';
+import { HttpClient, HttpClientModule } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { IRequestConfigExecute, RequestEventEnum } from '@waas/core';
+
+import { HttpTransportAdapterService } from './http-transport-adapter.service';
+import { WaasModule } from './waas.module';
+
+describe('HttpTransportAdapterService', () => {
+  let http: HttpTestingController;
+  let adapter: HttpTransportAdapterService;
+
+  beforeEach(() => {
+    TestBed.resetTestEnvironment();
+    TestBed.initTestEnvironment(BrowserDynamicTestingModule,
+      platformBrowserDynamicTesting());
+
+    TestBed.configureTestingModule({
+      imports: [WaasModule, HttpClientModule, HttpClientTestingModule]
+    });
+    http = TestBed.get(HttpTestingController);
+    const client = TestBed.get(HttpClient);
+    adapter = new HttpTransportAdapterService(client);
+  });
+
+  it('should be created', () => {
+    expect(adapter).toBeTruthy();
+  });
+
+  it('should make GET request', () => {
+    const url = 'api/testing';
+    const data = {method: 'GET', url, before: (data1: IRequestConfigExecute) => data1, after: (data1) => data1};
+    adapter.executeRequest(data).subscribe(() => {});
+    const req = http.expectOne(url);
+    req.flush({data: 'some data'});
+    expect(req.request.method).toBe('GET');
+  });
+
+  it('should return response with meta', (done) => {
+    const url = 'api/testing';
+    const headers = {Authorization: 'Basic'};
+    const params = {id: '15'};
+    const status = 200;
+    const statusText = 'Ok';
```

Comment:
Could your organize via mock files?
Global situation for each test case

Reply:
done